### PR TITLE
`.ci/write-dockerfile.sh`: Quoting fix for `unminimize`

### PR DESCRIPTION
**Pull request by mkoeppe**
On docker-ubuntu, we attempted to remove the `unminimize` program after running it once, but this has failed because of  quoting mistake. https://github.com/mkoeppe/sage/actions/runs/9219405769/job/25364502464#step:9:311

Fixed here.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: .ci/write-dockerfile.sh [hunk 1]
```diff
@@ -67,7 +67,7 @@ case $SYSTEM in
                 cat <<EOF
 RUN if command -v unminimize > /dev/null; then  \
         (yes | unminimize) || echo "(ignored)"; \
-        rm -f "$(command -v unminimize)";       \
+        rm -f "\$(command -v unminimize)";       \
     fi
```
Comment (kwankyu) [inline .ci/write-dockerfile.sh:71]: Are you sure? The old line (without `\`) seems correct.

Comment (kwankyu) [reply .ci/write-dockerfile.sh:71]:
```bash
$ command -v unminimize
/usr/local/sbin/unminimize
$ rm -f "$(command -v unminimize)";  
rm: cannot remove '/usr/local/sbin/unminimize': Permission denied
$ rm -f "\$(command -v unminimize)";
$ command -v unminimize
/usr/local/sbin/unminimize
```

Comment (mkoeppe) [reply .ci/write-dockerfile.sh:71]: This appears as part of a "here document" (`cat <<EOF`...`EOF`, https://www.gnu.org/software/bash/manual/bash.html#Here-Documents). The bug is that the `$(...)` is expanded at the time of running the script (in the host); it needs to be expanded when Docker executes the RUN command in the container.

Comment (kwankyu) [reply .ci/write-dockerfile.sh:71]: Ah, okay. I failed to see the context. Thanks.

**Review by kwankyu — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thanks.

**Pull request merged**